### ISSUE TITLE
train(unisona): 2nd crystallization run on the >=24GB workstation - full corpus + 7.6B PLT QLoRA

### ISSUE THREAD

**Issue by alex-place** [alex-place/lantern-os#1829]
## Goal

Run the **2nd Unisona crystallization** on a workstation with **≥24 GB VRAM** — the two trainings the 8 GB 3070 physically cannot do. Run-1 (this box) is in flight: Ouro-1.4B QLoRA, 773-row seq-1536 subset, 300 steps → `D:/lantern-train/unisona-v1-adapters`.

## Why a 2nd box

- The corpus has **1,178 clean rows** but only **530 fit seq=1536** — the 648 long PR-diff rows need seq≈4096, which swaps to CPU on 8 GB (39 min/step).
- The **7.6B keystone-sigma0-plt QLoRA** (`models/keystone-sigma0-plt/train_lora.py`) requires ≥24 GB (4-bit base + LoRA + grad-checkpoint through the 2-loop PLT forward). This is the real Unisona base candidate; the 1.4B run is the fast-feedback probe.
- Cloud alternates are blocked (#1189: Kaggle dataset not attachable, Lightning 403), so a physical ≥24 GB workstation is the unblocked path.

## Tasks (in order)

1. **Corpus**: either copy `data/csf/unisona-crystallization.csf` (~1.8 MB, gitignored — transfer out-of-band) and unpack via `python -c "import sys; sys.path.insert(0,'src'); from csf import csf_pack; open('data/training/unisona-corpus.clean.jsonl','wb').write(csf_pack.read_file('data/csf/unisona-crystallization.csf','training/unisona-corpus.clean.jsonl'))"`, **or** regenerate with `scripts/pr_crystallize.py` (PR #1807): self repo + `--repo` externals (requests/express/pandas/flask/transformers/react/fastapi), then `decontaminate_training.py`.
2. **Ouro-1.4B run-2 (full corpus, long rows)**: `python scripts/train-qlora-ouro.py --base ByteDance/Ouro-1.4B --data data/training/unisona-corpus.clean.jsonl --out <out> --seq 4096 --epochs 2` (24 GB fits seq=4096; audit tokenizer p99 first).
3. **Stage-0 true parity for the PLT** (blocking precondition per ADR-0011 / #1743): capture reference logits with the upstream vLLM fork, then `models/keystone-sigma0-plt/check_parity.py --model D:/keystone-sigma0-plt-ckpt --dtype bf16 --ref ref_logits.pt`. Do not train the PLT until PASS.
4. **7.6B PLT QLoRA** on the same corpus: `models/keystone-sigma0-plt/train_lora.py` (LoRA q/k/v/o/gate/up/down, grad-checkpoint mandatory). Gotcha: loading via AutoModel needs `PYTHONPATH=<checkpoint dir>` (sibling `configuration_keystone_plt` import trips transformers `check_imports`).
5. **Eval gate (External Reality Rule)**: `scripts/eval_sigma0_adapter.py --base-model <base> --adapter <out> --4bit` on the 6-task canonical set + no-evidence probes; append to `data/eval/`. Baselines to beat: **PLT base pass@1 0.50 / ECE 0.47 / Σ₀-format 0.0** (`data/eval/keystone-plt-baseline.jsonl`); prior Ouro adapters 0.00–0.05 on held-out HumanEval. Adapters stay `verified:false` in `lib/local-model-registry.js` until a reproduced head-to-head win.

## Success criteria

A leaderboard/eval row per adapter with pass@1, ECE, format-adherence, abstention — win or lose, logged as a convergence record. Nothing flips `verified:true` without the on-box eval win.

Refs: PR #1807 (extractor), #1743 (PLT Stage-0), #1189 (cloud blocked), #1777 (Ouro capability bar), ADR-0010/0011.
